cachi2/interface/cli.py

```diff
@@ -0,0 +1,153 @@
+import importlib.metadata
+import json
+import logging
+from itertools import chain
+from pathlib import Path
+from typing import Optional, Union
+
+import pydantic
+import typer
+from typer import Option
+
+from cachi2.core.models.input import Request
+from cachi2.core.package_managers import gomod
+from cachi2.interface.logging import LogLevel, setup_logging
+
+app = typer.Typer()
+log = logging.getLogger(__name__)
+
+DEFAULT_SOURCE = "."
+DEFAULT_OUTPUT = "./cachi2-output"
+
+
+def version_callback(value: bool) -> None:
+    """If --version was used, print the cachi2 version and exit."""
+    if value:
+        print("cachi2", importlib.metadata.version("cachi2"))
+        raise typer.Exit()
+
+
+@app.callback()
+def cachi2(  # noqa: D103; docstring becomes part of --help message
+    version: bool = Option(
+        False,
+        "--version",
+        callback=version_callback,
+        is_eager=True,
+        help="Show version and exit.",
+    ),
+    log_level: LogLevel = Option(
+        "DEBUG",
```

Comment:
I think that, for a better user experience, we should leave the default log leve as INFO, since it creates a more concise output.

Reply:
> Also, we should aim to improve the logs, as they are the primary indicator of progress in a request now.

+1, I'll file a story for that and change the default to INFO